File: okhttplib/src/main/java/com/example/okhttplib/view/RToolbar.java
package com.example.okhttplib.view;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import android.util.AttributeSet;
import android.view.View;

import com.blankj.utilcode.util.ActivityUtils;
import com.example.okhttplib.base.RBaseActivity;

public class RToolbar extends Toolbar {
    public RToolbar(Context context) {
        this(context, null);
    }

    public RToolbar(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        setNavigationOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                RBaseActivity topActivity = (RBaseActivity) ActivityUtils.getTopActivity();
                topActivity.finishActivity();
            }
        });
    }
}

File: okhttplib/src/main/java/com/example/okhttplib/view/RToolbar.kt
package com.example.okhttplib.view

import android.content.Context
import androidx.appcompat.widget.Toolbar
import android.util.AttributeSet
import android.view.View

import com.blankj.utilcode.util.ActivityUtils
import com.example.okhttplib.base.RBaseActivity
import com.example.okhttplib.utils.ActivityManager
import com.wuyr.activitymessenger.ActivityMessenger

class RToolbar @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : Toolbar(context, attrs) {

    init {
        setNavigationOnClickListener {
            ActivityManager.getInstance().finishActivity()
        }
    }
}
